packages/ses/test/import-cjs.test.js

```diff
@@ -607,6 +607,65 @@ test('synchronous loading via importNowHook', async t => {
   compartment.importNow('./main.js');
 });
 
+const importNowHookWithCyclicDep = specifier => {
+  if (specifier === './a.cjs') {
+    return CjsModuleSource(
+      `
+      exports.a = 42;
+      exports.B = require('./b.cjs');
+    `,
+      'https://example.com/a.cjs',
+    );
+  }
+  if (specifier === './b.cjs') {
+    return CjsModuleSource(
+      `
+      exports.b = 44
+      exports.A = require('./a.cjs');
+    `,
+      'https://example.com/b.cjs',
+    );
+  }
+  throw Error(`Cannot load module for specifier ${specifier}`);
+};
+
+test('import handles a cycle in CommonJS modules', async t => {
+  t.plan(4);
+  const compartment = new Compartment({
+    resolveHook: resolveNode,
+    importHook: async a => importNowHookWithCyclicDep(a),
+    importNowHook: importNowHookWithCyclicDep,
+    __options__: true,
```

Comment:
Out of curiosity, what does this do?

Reply:
It's the new compartment options syntax. Compartment used to accept 3 objects in the original signature. This is a gradual migration thing.